lib/membrane_rtc_engine/engine.ex

```diff
@@ -517,48 +508,143 @@ defmodule Membrane.RTC.Engine do
     {{:ok, tracks_msgs}, state}
   end
 
-  defp link_inbound_track({track_id, encoding}, tee, waiting_for_linking, ctx) do
-    reduce_children(ctx, {[], waiting_for_linking}, fn
-      {:endpoint, endpoint_id}, {new_links, waiting_for_linking} ->
-        if MapSet.member?(waiting_for_linking[endpoint_id], track_id) do
-          new_link =
-            link(tee)
-            |> via_out(:copy)
-            |> via_in(Pad.ref(:input, track_id), options: [encoding: encoding])
-            |> to({:endpoint, endpoint_id})
-
-          waiting_for_linking =
-            Map.update!(waiting_for_linking, endpoint_id, &MapSet.delete(&1, track_id))
-
-          {new_links ++ [new_link], waiting_for_linking}
-        else
-          {new_links, waiting_for_linking}
+  defp inbound_track_specs({track_id, encoding}, track_endpoint_node, tee, ctx, state) do
+    reduce_children(ctx, {[], [], state}, fn
+      {{:endpoint, endpoint_id}, child_entry}, {local_links, actions, state} ->
+        other_endpoint_node = node(child_entry.pid)
+        track_source_id = {other_endpoint_node, track_id}
+        track_source = {:track_source, track_source_id}
+
+        cond do
+          track_endpoint_node == other_endpoint_node and
+              MapSet.member?(state.waiting_for_linking[endpoint_id], track_id) ->
+            # Link track to endpoints on the same node
+            new_links = [
+              link(tee)
+              |> via_out(:copy)
+              |> via_in(Pad.ref(:input, track_id), options: [encoding: encoding])
+              |> to({:endpoint, endpoint_id})
+            ]
+
+            waiting_for_linking =
+              Map.update!(state.waiting_for_linking, endpoint_id, &MapSet.delete(&1, track_id))
+
+            {local_links ++ new_links, actions,
+             %{state | waiting_for_linking: waiting_for_linking}}
+
+          MapSet.member?(state.waiting_for_linking[endpoint_id], track_id) and
+              state.cross_node_track_receivers[track_source_id] ->
+            # Link track to existing UDP receiver on remote node
+            new_spec = %ParentSpec{
+              links: [
+                link(track_source)
+                |> via_in(Pad.ref(:input, track_id), options: [encoding: encoding])
+                |> to({:endpoint, endpoint_id})
+              ],
+              node: other_endpoint_node
+            }
+
+            {local_links, actions ++ [spec: new_spec], state}
+
+          MapSet.member?(state.waiting_for_linking[endpoint_id], track_id) ->
+            # Link track to new UDP receiver on remote node
+            new_spec = %ParentSpec{
+              children: %{track_source => Membrane.NodeProxy.Source},
+              links: [
+                link(track_source)
+                |> via_in(Pad.ref(:input, track_id), options: [encoding: encoding])
+                |> to({:endpoint, endpoint_id}),
+                link(track_source)
+                |> via_out(:data_channel)
```

Comment:
The data channel is used by the `track_source` to provide the `track_sink` with its addresses and port, so that the sink can do a quick handshake and then start sending to the correct node. If the `:data_channel` was the opposite direction, the sink would never be notified about the new destination.